core/parentchain/indirect-calls-executor/src/executor.rs

```diff
@@ -144,7 +150,13 @@ impl<
 		let filter_events = events.get_transfer_events();
 
 		if let Ok(events) = filter_events {
-			events.iter().for_each(|event| info!("Found transfer_event: {:?}", event))
+			events
+				.iter()
+				.filter(|&event| event.to == PrivacySidechain::SHIELDING_ACCOUNT)
+				.try_for_each(|event| {
+					info!("transfer_event: {}", event);
+					PrivacySidechain::shield_funds(&event.from, event.amount)
+				})?;
 		}
```

Comment:
This part can then be more generic. I want this simply to be: `ParentchainEventHandler::handle_events()`